files/scripts/syncd.sh

```diff
@@ -28,6 +28,10 @@ function startplatform() {
         debug "Firmware update procedure ended"
     fi
 
+    PLATFORM_DIR=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
```

Comment:
why not use $sonic_asic_platform?

Reply:
if we want to do this one, we should inject the hook for all functions startplatform, waitplatform, stopplatform1, and stopplatform2.

Reply:
and we should also refactor the existing code to use this platform hook.

Reply:
> why not use $sonic_asic_platform?

$sonic_asic_platform does not provide platform agnostic approach to otherwise a Platform Independent SONiC code under files/scripts path. Given that this is a run time check, using DEVICE_METADATA seems more as a generic approach

Reply:
> and we should also refactor the existing code to use this platform hook.

I agree with refactoring the existing code. The whole code has vendor hooks and is not maintainable. I can look into that approach. 

Can that be considered as a different exercise given we have dependencies with other vendors using this code.

Reply:
> startplatform, waitplatform, stopplatform1, and stopplatform2.

Presently, we need platform hooks for startplatform() API only. This is an easily extendable hook for any other function. We see an opportunity to restructure files/scripts/syncd.sh to pursue the following instead of making vendor specific calls from each function: 



function startplatform () {
    F="/usr/share/sonic/device/$PLATFORM_DIR/scripts/startplatform_routines.sh"
    [ -x $F ] && $F start $DEV
}

function stopplatform() {
    F="/usr/share/sonic/device/$PLATFORM_DIR/scripts/stopplatform_routines.sh"
    [ -x $F ] && $F start $DEV
}

...
PLATFORM_DIR=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`

case "$1" in
    start|wait|stop)
        $1
        ;;
    *)
        echo "Usage: $0 {start|wait|stop}"
        exit 1
        ;;
esac